From 78ae92a82d626d7da998a88412a2d39358d5c2a5 Mon Sep 17 00:00:00 2001
From: Daniel Albers <daniel@lbe.rs>
Date: Fri, 31 May 2013 14:39:34 +0200
Subject: [PATCH 127/160] journal: take KeepFree into account when reporting
 maximum size

When reporting the maximum journal size add a hint if it's limited
by KeepFree.

(cherry picked from commit fe1abefcd3bf1718dde3b5b835db56142c9f7082)
---
 src/journal/journald-server.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/journal/journald-server.c b/src/journal/journald-server.c
index da5b725..0bf557c 100644
--- a/src/journal/journald-server.c
+++ b/src/journal/journald-server.c
@@ -780,6 +780,9 @@ static int system_journal_open(Server *s) {
         char *fn;
         sd_id128_t machine;
         char ids[33];
+        uint64_t avail;
+
+        avail = available_space(s);
 
         r = sd_id128_get_machine(&machine);
         if (r < 0)
@@ -821,6 +824,10 @@ static int system_journal_open(Server *s) {
                         server_driver_message(s, SD_ID128_NULL, "Allowing system journal files to grow to %s.",
                                               format_bytes(fb, sizeof(fb), s->system_metrics.max_use));
 
+                        if (s->system_metrics.max_use > avail)
+                               server_driver_message(s, SD_ID128_NULL, "Journal size currently limited to %s due to SystemKeepFree.",
+                                                     format_bytes(fb, sizeof(fb), avail));
+
                 } else if (r < 0) {
 
                         if (r != -ENOENT && r != -EROFS)
@@ -874,6 +881,10 @@ static int system_journal_open(Server *s) {
                         server_fix_perms(s, s->runtime_journal, 0);
                         server_driver_message(s, SD_ID128_NULL, "Allowing runtime journal files to grow to %s.",
                                               format_bytes(fb, sizeof(fb), s->runtime_metrics.max_use));
+
+                        if (s->system_metrics.max_use > avail)
+                               server_driver_message(s, SD_ID128_NULL, "Journal size currently limited to %s due to RuntimeKeepFree.",
+                                                     format_bytes(fb, sizeof(fb), avail));
                 }
         }
 
-- 
1.8.3.1

